partiql-lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -227,12 +223,28 @@ internal class CompilerPipelineImpl(
     private val ion: IonSystem,
     private val parser: Parser,
     override val compileOptions: CompileOptions,
-    override val functions: Map<String, ExprFunction>,
+    override val functions: List<ExprFunction>,
     override val customDataTypes: List<CustomType>,
     override val procedures: Map<String, StoredProcedure>,
     private val preProcessingSteps: List<ProcessingStep>,
     override val globalTypeBindings: Bindings<StaticType>?
 ) : CompilerPipeline {
+
+    @Deprecated(
+        "Functions should be a list.",
+        level = DeprecationLevel.WARNING
+    )
+    constructor(
+        ion: IonSystem,
+        parser: Parser,
+        compileOptions: CompileOptions,
+        functionMap: Map<String, ExprFunction>,
```

Comment:
```suggestion
        functions: Map<String, ExprFunction>,
```

The signature needs to match the original.